File: frontend/src/types/syllabus.ts
// API Response types matching backend endpoints

export interface SubTopic {
  id: number;
  topic_id: number;
  content: string;
}

export interface Topic {
  id: number;
  number: string;
  title: string;
  unit_id?: number;
  unit_name: string;
  main_topic_name: string;
  sub_topics: SubTopic[];
}

export interface Unit {
  id: number;
  name: string;
  main_topic_id?: number;
  topics_count?: number;
  topics?: TopicBrief[];
}

export interface TopicBrief {
  id: number;
  number: string;
  title: string;
}

export interface MainTopic {
  id: number;
  name: string;
  sub_topic_count?: number;
  unit_count?: number;
  topic_count?: number;
  units?: Unit[];
}

export interface SearchResult {
  topic_id: number;
  topic_number: string;
  topic_title: string;
  unit_name: string;
  main_topic_name: string;
  matched_sub_topic: string | null;
}

export interface HealthResponse {
  status: string;
  app: string;
  version: string;
  database: string;
  llm_provider: string;
  llm_model: string;
}

File: backend/app/schemas/syllabus.py
"""Pydantic schemas for syllabus API responses."""

from pydantic import BaseModel


# ── Sub-Topic ─────────────────────────────────────────────────────
class SubTopicOut(BaseModel):
    id: int
    content: str

    model_config = {"from_attributes": True}


# ── Topic ─────────────────────────────────────────────────────────
class TopicBrief(BaseModel):
    """Compact topic without sub-topics (used in listings)."""
    id: int
    number: str
    title: str

    model_config = {"from_attributes": True}


class TopicDetail(BaseModel):
    """Full topic with all sub-topics."""
    id: int
    number: str
    title: str
    unit_name: str
    main_topic_name: str
    sub_topics: list[SubTopicOut]

    model_config = {"from_attributes": True}


# ── Unit ──────────────────────────────────────────────────────────
class UnitBrief(BaseModel):
    id: int
    name: str
    topic_count: int

    model_config = {"from_attributes": True}


class UnitDetail(BaseModel):
    id: int
    name: str
    topics: list[TopicBrief]

    model_config = {"from_attributes": True}


# ── Main Topic ────────────────────────────────────────────────────
class MainTopicBrief(BaseModel):
    id: int
    name: str
    unit_count: int
    topic_count: int

    model_config = {"from_attributes": True}


class MainTopicDetail(BaseModel):
    id: int
    name: str
    units: list[UnitDetail]

    model_config = {"from_attributes": True}


# ── Part Progress (for Curriculum grid) ────────────────────────────
class PartProgressItem(BaseModel):
    part_id: int
    completed: int
    total: int


# ── Search ────────────────────────────────────────────────────────
class SearchResult(BaseModel):
    topic_id: int
    topic_number: str
    topic_title: str
    unit_name: str
    main_topic_name: str
    matched_sub_topic: str | None = None
